code/modules/projectiles/projectile/bullets.dm

```diff
@@ -988,3 +988,17 @@ obj/item/projectile/bullet/suffocationbullet
 
 /obj/item/projectile/bullet/syringe/dart
 	stealthy = TRUE
+
+/obj/item/projectile/bullet/syringe/candycane
+	name = "Candycane"
+	icon_state = "candycane"
+	nodamage = 0
+	damage = 25
+	capacity = 20
+	decay_type = null
+	custom_impact = null
+
+/obj/item/projectile/bullet/syringe/candycane/New()
+	..()
+	reagents.add_reagent(CARAMEL, 19)
+	reagents.add_reagent(MINTTOXIN, 1) //I feel a little bad, might make it a random thing between this and normal mint or sugar. It's fun if you make them fat with the gun not so fun if you one shot fatsec
```

Comment:
I think the idea of this gun is fun but the implementation is bad. The caramel will take so long to work through the target's body that this is not practical for its intended purpose **even if the person was already full**. On the other hand, any amount of minttoxin will gib someone.

Here's what I would do:
1) make a new molasses reagent (key in gingerbreads) that directly increases overeatduration by a ton, like 212 per shot. You need to get them to 500 to fattify them (and they will lose 2 per tick for not actually being overstuffed).
2) change minttoxin so it will only gib if you have at least 1u in your system, and change this gun to give 0.8 per shot (remember this, too, will decay at a rate of 0.2 per tick) - this means one shot won't kill, and to finish them you need to hit them twice within 8 seconds

Reply:
I thought its intended purpose was to spread jolly mood by making people happy and well fed until a completely unforeseen and terrible accident happens. In which case the joke would be on the wielder.

Reply:
Little of both. The intention was a gun that you wouldn't use on people for raw damage but still helps you get through the vault. Wanted to make something more interesting than high numbers on bullets and fit the theme of "Diabetes gun" 
Changed it to inject "diabeetusol". The gun injects 10u and the whip injects 2u. 
5-10u makes you dizzy, confused, and rarely fall asleep for a second. Essentially it makes you hypo/hyperglycemic 
10-25u makes you rapidly gain weight and hear trombone sounds. Two shots will make you maxfat
25u+ will give you a heart attack nearly identical to corn oil. I meant to have it only do so if you were fat but I just couldn't get the interaction to work properly. Besides if you have 25u in you you're going to be fat. 

I like this more since it doesn't gib you and makes it easier to get you back into the game, but it's still a fun weapon if a traitor somehow got their hands on it.